packages/plugins/src/foundry/utils/index.ts

```diff
@@ -1063,16 +1066,35 @@ export const convertLibraryFormat = (
     return `${filePath}:${contractName}=${ethers.getAddress(address)}`
   })
 }
+
+const toSphinxTransactionEstimatedGas = (
+  response: SphinxTransactionResponse
+): EstimateGasTransactionData => {
+  return {
+    to: response.to,
+    from: response.from,
+    data: response.data,
+    gasLimit: response.gasLimit.toString(),
+    gasPrice: response.gasPrice.toString(),
+    value: response.value.toString(),
+    chainId: response.chainId.toString(),
+  }
+}
+
 /**
  * Estimates the gas used by a deployment on a single network. Includes a buffer of 30% to account
  * for variations between the local simulation and the production environment. Also adjusts the
  * minimum gas limit on networks like Arbitrum to include the L1 gas fee, which isn't captured on
  * forks.
  */
 export const getEstimatedGas = async (
-  receipts: Array<SphinxTransactionReceipt>,
+  transactions: SimulationTransactions,
```

Comment:
Generally, it'd be helpful to add unit tests for functions like this that contain some nontrivial logic. It's hard to feel confident as a reviewer that this is bug free, even if I attempt to mentally walk through the code. It's fine not to add unit tests for this function, but generally, I think this'd help reduce the chance of bugs and make reviews easier